lib/elixir/lib/kernel.ex

```diff
@@ -3087,6 +3087,16 @@ defmodule Kernel do
           {expanded, nil}
       end
 
+    # We do this so that the block is not tail-call optimized and stacktraces
+    # are not messed up. Basically, we just insert something between the return
+    # value of the block and what is returned by defmodule. Using just ":ok" or
+    # similar doesn't work because it's likely optimized away by the compiler.
+    block = quote do
+      result = unquote(block)
+      (fn -> :ok end).()
```

Comment:
@josevalim I initially went with something like your suggestion and used `:elixir_utils.noop()`, but I can see no downsides of just using an anon function like this, but I can see the upside of not having to define/export an additional function. Wdyt?